### PR TITLE
Remove hard-coded list of legal Imgix URL params

### PR DESCRIPTION
**Pull request by hashknot**
PR #55 laid the groundwork to support arbitrary URL params in `ix_image_tag` and `ix_picture_tag`. This PR removes `parameters.json` and the code associated with segregating imgix and non-imgix URL params. Through this PR, we won't need to update the library every time a new imgix URL parameter is introduced.

This is a **backwards-incompatible** change.

### CODE REVIEW

File: lib/imgix/rails/picture_tag.rb [hunk 1]
```diff
@@ -18,10 +18,9 @@ def render
         source_tag_opts = opts[:tag_options] || {}
         source_url_params = opts[:url_params] || {}
         widths = opts[:widths]
-        if opts.except(:tag_options, :url_params, :widths).length > 0
-          ActiveSupport::Deprecation.warn('use :tag_options, :url_params, :widths instead.')
-          source_tag_opts = opts.slice!(*self.class.available_parameters).except(:widths)
-          source_url_params = opts.slice(*self.class.available_parameters).except(:widths)
+        unsupported_opts = opts.except(:tag_options, :url_params, :widths)
```
Comment (jayeb) [inline lib/imgix/rails/picture_tag.rb]: Can we pull this out into a separate PR?

File: lib/imgix/rails/view_helper.rb [hunk 2]
```diff
@@ -8,36 +8,11 @@ module Rails
...
-
-          if options.except(:tag_options, :url_params, :widths).length > 0
-            ActiveSupport::Deprecation.warn('options Hash is deprecated; use :tag_options, :url_params, :widths instead.')
-            tag_options = options.slice!(*Imgix::Rails::Tag.available_parameters).except(:widths)
-            url_params = options.slice(*Imgix::Rails::Tag.available_parameters).except(:widths)
-          end
-
-          Imgix::Rails::ImageTag.new(source, tag_options: tag_options, url_params: url_params, widths: widths).render
-        end
+      def ix_image_tag(source, tag_options: {}, url_params: {}, widths: [])
```
Comment (jayeb) [inline lib/imgix/rails/view_helper.rb]: Wow, so much cleaner 👍

File: lib/imgix/rails/view_helper.rb [hunk 3]
```diff
@@ -8,36 +8,11 @@ module Rails
...
+        Imgix::Rails::ImageTag.new(source, tag_options: tag_options, url_params: url_params, widths: widths).render
       end
 
-      def ix_picture_tag(source, picture_tag_options: {}, imgix_default_options: {}, breakpoints:, tag_options: {}, url_params: {})
```
Comment (jayeb) [inline lib/imgix/rails/view_helper.rb]: Same as above--can we pull all of the code that's just removing deprecated things out into a separate PR? We can bundle the two so they're merged and released together, but I think they should be kept separate.

**Comment by jayeb** [comment]
It's also worth noting that this *adds* support for non-standard imgix parameters, which are legal in imgix URLs (We just ignore them) but previously were being rejected.

File: lib/imgix/rails/picture_tag.rb (hunk 1 above)
Comment (hashknot) [reply lib/imgix/rails/picture_tag.rb]: This is heavily tied to the legal parameters code. If keys other than these three are specified, they used to be separated into URL params/tag options based on the list of legal parameters. Because we don't have a list of legal parameters anymore, segregation isn't possible. So this code raises an error with a list of keys that aren't supported.

File: lib/imgix/rails/view_helper.rb (hunk 3 above)
Comment (hashknot) [reply lib/imgix/rails/view_helper.rb]: I will move this into a separate PR.

**Comment by jayeb** [comment]
Augh, I totally forgot about that. Ignore me!

File: lib/imgix/rails/picture_tag.rb (hunk 1 above)
Comment (jayeb) [reply lib/imgix/rails/picture_tag.rb]: Right, okay. I understand now.
